partitioner/src/main/java/io/confluent/connect/storage/partitioner/TimeBasedPartitioner.java

```diff
@@ -234,14 +234,107 @@ public void configure(Map<String, Object> config) {
       dateTime = ISODateTimeFormat.dateTime();
     }
 
+    private Object getField(Struct struct, String fieldName) {
+      Object field;
+      try {
+        field = struct.get(fieldName);
+      } catch (DataException e) {
+        throw new DataException(
+                String.format("The field named '%s' does not exist.", fieldName), e);
+      }
+      return field;
+    }
+
+    private Object getNestedFieldValue(Struct struct) {
+      final String[] fieldNames = fieldName.split("\\.");
+      Struct tmpStruct = struct;
+      Object tmpObject;
+      try {
+        // Iterate down to final struct
+        int i = 0;
+        while (i < fieldNames.length - 1) {
+          try {
+            tmpObject = getField(tmpStruct, fieldNames[i]);
+          } catch (DataException e) {
+            throw new DataException(
+                    String.format("Unable to find nested field '%s'", fieldNames[i]));
+          }
+          tmpStruct = (Struct) tmpObject;
+          i++;
+        }
+        // Extract from final struct
+        tmpObject = getField(tmpStruct, fieldNames[i]);
+      } catch (DataException e) {
+        throw new DataException(
+                String.format("The nested field named '%s' does not exist.", fieldName), e);
+      }
+      return tmpObject;
+    }
+
+    private Object getNestedFieldValue(Map<?, ?> valueMap) {
+      final String[] fieldNames = fieldName.split("\\.");
+      Map<?, ?> tmpMap = valueMap;
+      Object tmpObject;
+      try {
+        // Iterate down to final map
+        int i = 0;
+        while (i < fieldNames.length - 1) {
+          tmpObject = tmpMap.get(fieldNames[i]);
+          if (tmpObject == null) {
+            throw new DataException(
+                    String.format("Unable to find nested field '%s'", fieldNames[i]));
+          }
+          tmpMap = (Map<?, ?>) tmpObject;
+          i++;
+        }
+        // Extract from final map
+        tmpObject = tmpMap.get(fieldNames[i]);
+        if (tmpObject == null) {
+          throw new DataException(
+                  String.format("Unable to find nested field '%s'", fieldNames[i]));
+        }
+      } catch (DataException e) {
+        throw new DataException(
+                String.format("The nested field named '%s' does not exist.", fieldName), e);
+      }
+      return tmpObject;
+    }
+
+    private Field getNestedField(Schema schema) {
+      final String[] fieldNames = fieldName.split("\\.");
+      if (fieldNames.length == 1) {
+        return schema.field(fieldName);
```

Comment:
Wasn't sure if the other `getNested` methods should return early like this. I did it here because I was getting IndexOutOfBounds, and didn't want to deal with the correct logic in the while loop.